### PR TITLE
[ULC] added file completion after :load commands

### PR DESCRIPTION
**Pull request by colmbaston**

### CODE REVIEW

**Comment by colmbaston** [comment]
**TODO**:

- Use `" ()λ\\."` as word separators in tab-completion for terms, but only use `" "` for filepaths.
- Parsing and printing filepaths with escape characters in a platform-independent way.

File: ULC/Repl.hs [hunk 1]
```diff
@@ -45,10 +46,12 @@ settings :: Settings (StateT Environment IO)
 settings = Settings contextCompletion Nothing True
   where
     contextCompletion :: CompletionFunc (StateT Environment IO)
```
Comment (lukeg101) [inline ULC/Repl.hs:48]: can you add a quick comment for each of these please, it will help those who see this in the future

File: ULC/Parser.hs [hunk 2]
```diff
@@ -178,7 +178,7 @@ parseReplCommand s = case parse pCommand s of
 
 -- | Parse a command prefixed by a colon, or parse a raw term.
 pCommand :: Parser Command
-pCommand = pReds +++ pLet +++ pLoad +++ (T <$> pTerm) +++ pComment
+pCommand = spaces (pReds +++ pLet +++ pLoad +++ (T <$> pTerm) +++ pComment)
```
Comment (lukeg101) [inline ULC/Parser.hs:181]: why do you declare pReds but not `T <$> pTerm`?
For uniformity this would look better

File: ULC/Repl.hs [hunk 3]
```diff
@@ -51,7 +51,7 @@ settings = Settings contextCompletion Nothing True
     completions :: String -> String -> StateT Environment IO [Completion]
     completions prev xs = if "daol:" `isPrefixOf` dropWhile isSpace prev -- check if previous word (in reverse) is ":load"
                             then listFiles xs
-                            else map (\x -> Completion x x True) . filter (xs `isPrefixOf`) . (comms ++) . M.keys <$> get
+                            else map (\x -> Completion x x True) . filter (xs `isPrefixOf`) . (if all isSpace prev then (comms ++) else id) . M.keys <$> get
```
Comment (lukeg101) [inline ULC/Repl.hs:54]: This is quite the long line, please can you split this up into something shorter

**Review by lukeg101 — CHANGES_REQUESTED** [review]
This is very good, comments inlined.
Some of the recent changes to ULC have adopted a long line length, I follow the belief that lines should be around [80 columns in width ](https://softwareengineering.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width).
Please can you ensure that all Haskell files in ULC adopt this convention (80 chars or less). This is mostly due to the Parser changes but for uniformity it would be nice if all of ULC followed this convention

**Pull request closed without merging**